src/main/java/com/synopsys/integration/detect/configuration/DetectConfigurationFactory.java

```diff
@@ -198,8 +198,11 @@ public DetectToolFilter createToolFilter(RunDecision runDecision, BlackDuckDecis
         AllNoneEnumCollection<DetectTool> excludedTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS_EXCLUDED);
         ExcludeIncludeEnumFilter<DetectTool> filter = new ExcludeIncludeEnumFilter<>(excludedTools, includedTools);
 
+        boolean rapidSignatureScanEnabled = includedTools.containsValue(DetectTool.SIGNATURE_SCAN) && 
+                blackDuckDecision.scanMode() == BlackduckScanMode.RAPID;
         boolean iacEnabled = includedTools.containsValue(DetectTool.IAC_SCAN) || !detectConfiguration.getValue(DetectProperties.DETECT_IAC_SCAN_PATHS).isEmpty();
```

Comment:
To avoid breaking the "excludes always win" pattern, this code should use the filter.shouldInclude() method instead of the includedTools list.